share/wake/lib/system/sources.wake

```diff
@@ -58,6 +58,42 @@ export def source (file: String): Result Path Error =
         x, Nil = Pass x
         _      = unreachable "a quoted RegExp can never match more than one distinct String"
 
+# Allows claiming of a file so long as the file is within
+# wakeroot. Keep in mind that if another job produces this
+# file, claim will fail. Likewise if you claim a file then
+# any job that produces this file will fail. Claim should
+# be thought of as a "virtual" job and wake demands that
+# all paths be produced by exactly one job, virtual or
+# otherwise.
+#
+# For source files please use `source`. For files outside
```

Comment:
I'm wondering if this should error if they try to claim a file that source could have provided

Reply:
At the time I wasn't sure if that was possible but thinking about it now it looks like I could se `prim "sources"` to do that. I can't use `source` or `sources` because those claim the file but `prim "sources"` just tells you which sources match which is better.

Reply:
This will be fixed by not using `raw_source` and "<source>".

Reply:
I think Matthew's point is that we're asking people to prefer `source` but not enforcing that. If we Instead used `prim "sources"` we could enforce it rather than leaving it for review.